modules/internal/CString.chpl

```diff
@@ -217,21 +217,21 @@ module CString {
   extern proc c_string_to_uint64_t(x:c_string) : uint(64);
 
   inline proc _cast(type t, x:c_string) where t == int(8)
-    return c_string_to_int8_t(x);
+    return c_string_to_int8_t((x:string).strip().c_str());
   inline proc _cast(type t, x:c_string) where t == int(16)
-    return c_string_to_int16_t(x);
+    return c_string_to_int16_t((x:string).strip().c_str());
   inline proc _cast(type t, x:c_string) where t == int(32)
-    return c_string_to_int32_t(x);
+    return c_string_to_int32_t((x:string).strip().c_str());
   inline proc _cast(type t, x:c_string) where t == int(64)
-    return c_string_to_int64_t(x);
+    return c_string_to_int64_t((x:string).strip().c_str());
   inline proc _cast(type t, x:c_string) where t == uint(8)
-    return c_string_to_uint8_t(x);
+    return c_string_to_uint8_t((x:string).strip().c_str());
   inline proc _cast(type t, x:c_string) where t == uint(16)
-    return c_string_to_uint16_t(x);
+    return c_string_to_uint16_t((x:string).strip().c_str());
   inline proc _cast(type t, x:c_string) where t == uint(32)
-    return c_string_to_uint32_t(x);
+    return c_string_to_uint32_t((x:string).strip().c_str());
   inline proc _cast(type t, x:c_string) where t == uint(64)
-    return c_string_to_uint64_t(x);
+    return c_string_to_uint64_t((x:string).strip().c_str());
```

Comment:
I think we don't have a test case for the uint cases yet, can we add that to the other pull request?